src/main/java/edu/cornell/kfs/concur/businessobjects/ValidationResult.java

```diff
@@ -43,15 +44,25 @@ public void addMessage(String message) {
         if (messages == null) {
             messages = new ArrayList<String>();
         }
-        messages.add(message);
+        if (StringUtils.isNotBlank(message) && isNotDuplicateMessage(message)) {
+            messages.add(message);
+        }
+    }
+    
+    private boolean isNotDuplicateMessage(String message) {
+        for (String currentMessage : getMessages()) {
+            if (StringUtils.equals(currentMessage, message)) {
```

Comment:
The messages are built by validation service, so we shouldn't have a case sensitivity issue, but it won't hurt to put the check in.